### PR TITLE
fix(core): Fix missing maxRemainingAsgs

### PR DESCRIPTION
**Pull request by alanmquach**
There were 2 places where we failed to hydrate the `redblack` strategy with it's standard parameters (one of which is maxRemaningAsgs):

1. When adding a new cluster configuration but opting to copy the configuration from an existing server group, we were simply setting `command.strategy = 'redblack'` without running the `initializationMethod`
2. When changing from a strategy other than `none` to `redblack`, the `initializationMethod` actually gets run, however we set `maxRemaningAsgs` to `2` if `!command.strategy`, but that does not guarantee that `command.maxRemainingAsgs` has a value.

### CODE REVIEW

File: app/scripts/modules/core/src/deploymentStrategy/strategies/redblack/redblack.strategy.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ DeploymentStrategyRegistry.registerStrategy({
       rollback: {
         onFailure: false,
       },
-      maxRemainingAsgs: !command.strategy ? 2 : command.maxRemainingAsgs,
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/deploymentStrategy/strategies/redblack/redblack.strategy.ts]: Can you verify that this Boolean logic is correct?

Comment (alanmquach) [reply app/scripts/modules/core/src/deploymentStrategy/strategies/redblack/redblack.strategy.ts]: The old logic? I imagine it was there to preserve the value if someone were to switch strategies then switch back

Comment (alanmquach) [reply app/scripts/modules/core/src/deploymentStrategy/strategies/redblack/redblack.strategy.ts]: Oh but I suppose `!command.strategy` was probably never very correct to start. `command.maxRenainingAsgs === undefined` is probably more appropriate

**Comment by marchello2000** [comment]
❤️

File: app/scripts/modules/core/src/deploymentStrategy/strategies/redblack/redblack.strategy.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ DeploymentStrategyRegistry.registerStrategy({
...
+      maxRemainingAsgs: command.maxRemainingAsgs !== undefined ? command.maxRemainingAsgs : 2,
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/deploymentStrategy/strategies/redblack/redblack.strategy.ts]: wondering if you're intentionally excluding `null` check, or if we can do:

```suggestion
      maxRemainingAsgs: command.maxRemainingAsgs ?? 2,
```

Comment (alanmquach) [reply app/scripts/modules/core/src/deploymentStrategy/strategies/redblack/redblack.strategy.ts]: Oh right! That's a thing we can do now! Nope, not intentional.

**Pull request merged**